Block/Adminhtml/Customer/Tab/Customer.php

```diff
@@ -0,0 +1,200 @@
+        \Magento\Customer\Model\CustomerFactory $customerFactory,
+        \Magento\Framework\Registry $coreRegistry,
+        \Magento\Framework\App\RequestInterface $request,
+        \DevStone\UsageCalculator\Model\ResourceModel\UsageCustomer\CollectionFactory $collectionFactory,
+        array $data = [],
+        Visibility $visibility = null,
+        Status $status = null
+    ) {
+        $this->customerFactory = $customerFactory;
+        $this->coreRegistry = $coreRegistry;
+        $this->request = $request;
+        $this->collectionFactory = $collectionFactory;
+        $this->visibility = $visibility ?: ObjectManager::getInstance()->get(Visibility::class);
+        $this->status = $status ?: ObjectManager::getInstance()->get(Status::class);
+        parent::__construct($context, $backendHelper, $data);
+    }
+
+    /**
+     *
+     */
+    protected function _construct()
+    {
+        parent::_construct();
+        $this->setId('usage_calculator_customers');
+        $this->setDefaultSort('id');
+        $this->setDefaultFilter(['in_usage' => 1]);
+
+        $this->setUseAjax(true);
+    }
+
+    protected function _addColumnFilterToCollection($column)
+    {
+        if ($column->getId() == 'in_usage') {
+            $customerIds = $this->_getSelectedCustomer();
+            if (empty($customerIds)) {
+                $customerIds = 0;
+            }
+            $linkField = 'entity_id';
+            $filter    = $column->getFilter();
+            if ($filter !== false && $column->getFilter()->getValue()) {
+                $this->getCollection()->addFieldToFilter($linkField, ['in' => $customerIds]);
+            } elseif (!empty($customerIds)) {
+                $this->getCollection()->addFieldToFilter($linkField, ['nin' => $customerIds]);
+            }
+        } else {
+            parent::_addColumnFilterToCollection($column);
+        }
+
+        return $this;
+    }
+
+
+    /**
+     * @return Extended
+     */
+    protected function _prepareCollection()
+    {
+        $collection = $this->customerFactory->create()->getCollection()->addAttributeToSelect('*');
+        $storeId = (int)$this->getRequest()->getParam('store', 0);
+        if ($storeId > 0) {
+            $collection->addStoreFilter($storeId);
+        }
+
+        $this->setCollection($collection);
+
+        return parent::_prepareCollection();
+    }
+
+    /**
+     * @return Extended
+     * @throws \Exception
+     */
+    protected function _prepareColumns()
+    {
+        $this->addColumn(
+            'in_usage',
+            [
+                'type' => 'checkbox',
+                'name' => 'in_usage',
+                'values' => $this->_getSelectedCustomer(),
+                'index' => 'entity_id',
+                'header_css_class' => 'col-select col-massaction',
+                'column_css_class' => 'col-select col-massaction'
+            ]
+        );
+
+        $this->addColumn(
+            'entity_id',
```

Comment:
No need to display customer ID here

Reply:
customer id for sorting purpose if you want we can remove customer id from grid

Reply:
I guess it's fine to keep it for sorting.